src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/HandshakeProtocol.cs

```diff
@@ -28,6 +28,7 @@ public static void WriteRequestMessage(HandshakeRequestMessage requestMessage, S
                 writer.WritePropertyName(ProtocolPropertyName);
                 writer.WriteValue(requestMessage.Protocol);
                 writer.WriteEndObject();
+                writer.Flush();
```

Comment:
It was happening on Close which was being called by JsonTextReader but I disabled it because PipeWriterStream threw an error on Flush. No longer using PipeWriterStream so this could be undone.